autometa/common/exceptions.py

```diff
@@ -150,7 +113,16 @@ def __str__(self):
         str
             Message written alongside raising the exception
         """
-        return self.value
+        message = """
+
+        NCBI databases nodes.dmp, names.dmp and merged.dmp are out of sync with each other.
```

Comment:
Are these out of sync with each other or out of sync with other NCBI databases, or both?

Reply:
What about `prot.accession2taxid`? Why would an error like this occur during the Autometa pipeline?

Reply:
As you mentioned correctly `DatabaseOutOfSyncError`  would happen if the `prot.accession2taxid.gz` is out of sync with `nodes`, `names` or `merged` . However, what about `nr.gz` being out of sync with `prot.accession2taxid.gz` as well? This may also raise errors right? I went to `diamond.py` and saw you have commented out the code that raises the `DatabasesOutOfDateError`. Should I uncomment it and replace this with the `DatabaseOutOfSyncError` ?  We can then display the original error message as well as the one raised by `exception.py` in which I can then include that all the 5 databases are out of sync with each other. Is that right?

Reply:
Could you link the lines to the code you are mentioning?

Reply:
These are the lines I was indicating to.
https://github.com/KwanLab/Autometa/blob/aaa7ff5c435be39fdaf95fe7c401e6f8a1789ee8/autometa/common/external/diamond.py#L528-L531
However, it seems I may not have the latest dev pulled.
In my local repository it is showing:
```python
        for sseqid in hit.sseqids:
            taxid = acc2taxids.get(sseqid)
            # if taxid is None:
            #     raise DatabasesOutOfDateError(f'{sseqid} deprecated/suppressed/removed')
            hit.sseqids[sseqid].update({"taxid": taxid})
    return hits
```
I was proposing to raise the `DatabaseOutOfSyncError` in case the `sseqid` from `nr.gz` is not found in the `prot.accession2taxid.gz` database. This would suggest that `nr.gz` is out of sync with `prot.accession2taxid.gz`. 
Upon raising the error then we can include the error message passed while raising the error as well as the error raised by `exceptions.py` in which we can then include that the databases are out of sync with each other and provide links to all 5. Is that right?

Reply:
I think something like this would be good. Also there exists a dead_prot.accession2taxid.gz file where we may be able to find these missing ids. (ftp://ftp.ncbi.nlm.nih.gov/pub/taxonomy/accession2taxid/dead_prot.accession2taxid.gz). It may be good to incorporate this into diamond.py or ncbi.py

Reply:
Great idea. I was thinking of incorporating this in `diamond.py`. However, I was thinking that can be separate PR in itself. Right now I have decided to raise the error in `diamond.py` and tell the user that all his databases are out of sync with each other. Is that okay?

Reply:
2 questions.
1. The reason for searching through dead_prot.accession2taxid.gz is that we can locate a taxid that is either deprecated/suppressed/removed and then locate its corresponding lastest taxid from merged.dmp. Is that right?
2. I think we should include `sys.exit(1)` to exit the pipeline, once the error is caught and logged.
https://github.com/KwanLab/Autometa/blob/aaa7ff5c435be39fdaf95fe7c401e6f8a1789ee8/autometa/__main__.py#L192-L194